tests/test_circuit_validation.py

```diff
@@ -288,26 +290,146 @@ def test_no_edges_h5():
                 del h5f['edges/default/' + ds]
+            h5f.create_group('edges/default/1')
         errors = test_module.validate(str(config_copy_path))
         assert errors == [Error(Error.FATAL, 'Population default of {} misses datasets {}'.
                                 format(edges_file, required_datasets))]
 
 
+def test_edge_population_multiple_groups():
+    with _copy_circuit() as (circuit_copy_path, config_copy_path):
+        edges_file = circuit_copy_path / 'edges.h5'
+        with h5py.File(edges_file, 'r+') as h5f:
+            h5f.create_group('edges/default/1')
+        errors = test_module.validate(str(config_copy_path), bbp_check=True)
+        assert BbpError(Error.WARNING, 'Population default of {} have multiple groups. '
+                                       'Cannot be read via bluepysnap or libsonata'.
+                        format(edges_file)) in errors
+
+
+def test_edge_population_missing_edge_group_id_one_group():
+    with _copy_circuit() as (circuit_copy_path, config_copy_path):
+        edges_file = circuit_copy_path / 'edges.h5'
+        with h5py.File(edges_file, 'r+') as h5f:
+            del h5f['edges/default/edge_group_id']
+        errors = test_module.validate(str(config_copy_path))
+        assert errors == [Error(Error.FATAL, 'Population default of {} misses dataset {}'.
+                                format(edges_file, {"edge_group_id"}))]
+
+
+def test_edge_population_missing_edge_group_index_one_group():
+    with _copy_circuit() as (circuit_copy_path, config_copy_path):
+        edges_file = circuit_copy_path / 'edges.h5'
+        with h5py.File(edges_file, 'r+') as h5f:
+            del h5f['edges/default/edge_group_index']
+        errors = test_module.validate(str(config_copy_path))
+        assert errors == [Error(Error.FATAL, 'Population default of {} misses dataset {}'.
+                                format(edges_file, {"edge_group_index"}))]
+
+
+def test_edge_population_missing_edge_group_id_index_one_group():
+    with _copy_circuit() as (circuit_copy_path, config_copy_path):
+        edges_file = circuit_copy_path / 'edges.h5'
+        with h5py.File(edges_file, 'r+') as h5f:
+            del h5f['edges/default/edge_group_index']
+            del h5f['edges/default/edge_group_id']
+        errors = test_module.validate(str(config_copy_path))
+        assert errors == []
+
+
+def test_edge_population_edge_group_different_length():
+    with _copy_circuit() as (circuit_copy_path, config_copy_path):
+        edges_file = circuit_copy_path / 'edges.h5'
+        with h5py.File(edges_file, 'r+') as h5f:
+            del h5f['edges/default/edge_group_index']
+            h5f.create_dataset('edges/default/edge_group_index', data=[0, 1, 2, 3, 4])
+        errors = test_module.validate(str(config_copy_path))
+        assert errors == [Error(Error.FATAL,
+                                'Population default of {} "edge_group_ids" and "edge_group_index" of different sizes'.
```

Comment:
Should be "edge_group_id" instead of "edge_group_id**s**" ?

Reply:
indeed !